test/framework/clusterctl/logger/logger.go

```diff
@@ -0,0 +1,146 @@
+// +build e2e
+
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logger
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+)
+
+// Provides a logr.Logger to use during e2e tests.
+
+type logger struct {
+	writer io.Writer
+	values []interface{}
+}
+
+var _ logr.Logger = &logger{}
+
+func (l *logger) Enabled() bool {
+	return true
+}
+
+func (l *logger) Info(msg string, kvs ...interface{}) {
+	values := copySlice(l.values)
```

Comment:
Can we use clusterctl/log package instead of adding a new one here?

Reply:
Unfortunately not with the current implementation of the logger. 
TBH  I don't even consider this as a viable option, because when I proposed an extensible implementation for the clusterctl logger, it was turned down...
But eventually we can converge later if there is agreement on this